llvm/test/CodeGen/LoongArch/ir-instruction/sdiv-udiv-srem-urem.ll

```diff
@@ -189,8 +189,6 @@ define signext i32 @sdiv_si32_ui32_ui32(i32 %a, i32 %b) {
 ;
 ; LA64-LABEL: sdiv_si32_ui32_ui32:
 ; LA64:       # %bb.0: # %entry
-; LA64-NEXT:    addi.w $a1, $a1, 0
```

Comment:
These two instructions should not be removed because the inputs are not guaranteed to be sign extended.

Reply:
The high 32 bits of input operands are ignored by `div.w`: https://loongson.github.io/LoongArch-Documentation/LoongArch-Vol1-EN.html

> DIV.W:
    quotient = signed(GR[rj][31:0]) / signed(GR[rk][31:0])
    GR[rd] = SignExtend(quotient[31:0], GRLEN)

Reply:
FYI RISC-V ISel doesn't generate `sext.w`: https://godbolt.org/z/af8vhhjeM

Reply:
The English version misses a description about {div,mod}.[wu] while the Chinese version has.
```
在 LoongArch64 位兼容的机器上,执行 DIV.W[U]和 MOD.W[U]指令时,如果通用寄存器 rj 或 rk 中的
数值超过了 32 位有符号数/无符号数的数值范围,则指令执行结果可以为无意义的任意值。
```
Note that "32位有符号数/无符号数" should be "32位有符号数".

This limitation means that we should ensure the input operands are signed extend 32-bits values.

What's more, LoongArch V1.10 introduced a CPUCFG bit indicating whether the above limitation exist or not. If this bit is `1`, we can omit the explict `addi.w $a1, $a1, 0`.

Chinese version: https://www.loongson.cn/download/index

Reply:
Fixed. I use `assertsexti32` instead. It only matches `AssertSExt` nodes.

Reply:
We should be more careful with division. for example:
```
define signext i32 @foo(i32 signext %a, i32 signext %b)  {
entry:                                                                                                           
  %conv = sext i32 %a to i64                                                                                     
  %conv1 = sext i32 %b to i64                                                                                    
  %div = sdiv i64 %conv1, %conv                                                                                  
  %conv2 = trunc i64 %div to i32                                                                                 
  ret i32 %conv2                                                                                                 
}                                                                                                                
```
In some cases this may not be possible using 32-bit division either.

Reply:
It has been proved by alive2: https://alive2.llvm.org/ce/z/AkH5Mp
IIRC LoongArch doesn't raise exceptions when signed overflow occurs (i.e., INT_MIN/-1).